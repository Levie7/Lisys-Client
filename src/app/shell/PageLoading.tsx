import React from 'react';

import { LoadingComponent } from 'src/core/chunk';

import { Spin } from 'src/shared/components/Spin';

export const PageLoading: LoadingComponent = ({ error }, props) => {
    if (error) {
        throw error;
    }
    return <Spin spinning={true}>{props.children}</Spin>;
};
