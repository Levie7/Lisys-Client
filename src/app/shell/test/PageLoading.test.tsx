import { shallow } from 'enzyme';
import React from 'react';

import { PageLoading } from '../PageLoading';

describe('PageLoading', () => {
    const props = {
        isLoading: false,
        pastDelay: false,
        timedOut: false,
        error: undefined,
        retry: jest.fn(),
    };
    const wrap = shallow(<PageLoading {...props} />);

    it('<PageLoading/> - should render pageLoading component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should contains memo spin component', () => {
        expect(wrap.find('Memo()').exists()).toBeTruthy();
    });

    it('should throw error', () => {
        let throwError;
        try {
            wrap.setProps({ error: 'error' });
        } catch (error) {
            throwError = error;
        }

        expect(throwError).toEqual('error');
    });
});
