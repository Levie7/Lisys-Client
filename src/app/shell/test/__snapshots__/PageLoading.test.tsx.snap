// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoading <PageLoading/> - should render pageLoading component correctly 1`] = `ShallowWrapper {}`;
